appveyor.yml

```diff
@@ -1,41 +1,57 @@
-image: Visual Studio 2017
+image:
+- Ubuntu
+- Visual Studio 2017
 
 skip_tags: false
 
 test: off
 
-init:
-- ps: >-
-    if ($env:APPVEYOR_REPO_TAG -eq "true")
-    {
-        Update-AppveyorBuild -Version ($env:APPVEYOR_REPO_TAG_NAME).TrimStart("v")
-    }
-
-after_build:
-- ps: >-
-    $wc = New-Object System.Net.WebClient
-
-    $wc.UploadFile("https://ci.appveyor.com/api/testresults/nunit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path "NetCoreClrTestResults.xml"))
-
-    $wc.UploadFile("https://ci.appveyor.com/api/testresults/nunit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path "NetCoreClrWeakNamedTestResults.xml"))
-
-    $wc.UploadFile("https://ci.appveyor.com/api/testresults/nunit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path "DesktopClrTestResults.xml"))
-
-    $wc.UploadFile("https://ci.appveyor.com/api/testresults/nunit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path "DesktopClrWeakNamedTestResults.xml"))
-
-on_success:
-- ps: >-
-    if ($env:APPVEYOR_REPO_TAG -eq "true")
-    {
-        nuget push ".\build\Castle.Core.${env:APPVEYOR_BUILD_VERSION}.nupkg" -ApiKey $env:NUGET_API_KEY -Source https://api.nuget.org/v3/index.json
-        nuget push ".\build\Castle.Core-log4net.${env:APPVEYOR_BUILD_VERSION}.nupkg" -ApiKey $env:NUGET_API_KEY -Source https://api.nuget.org/v3/index.json
-        nuget push ".\build\Castle.Core-NLog.${env:APPVEYOR_BUILD_VERSION}.nupkg" -ApiKey $env:NUGET_API_KEY -Source https://api.nuget.org/v3/index.json
-        nuget push ".\build\Castle.Core-Serilog.${env:APPVEYOR_BUILD_VERSION}.nupkg" -ApiKey $env:NUGET_API_KEY -Source https://api.nuget.org/v3/index.json
-    }
-
-build_script:
-- cmd: build.cmd
-
-artifacts:
-- path: build\*.nupkg
-  name: core
+for:
+-
+  matrix:
+    only:
+    - image: Ubuntu
+
+  build_script:
+  - uname -a
+  - mono --version
```

Comment:
In our Travis script we were installing Mono 5.16.0, is there a way we can pin the Mono version? Otherwise the chance that a new version will break tests is increased if Mono just gets upgraded on every build.

Reply:
1. I suppose AppVeyor don't update their VM images all that often, and they usually provide the previous image for some time in case breakages happen. Breaking changes would be a hassle, but it might be good for the Mono ecosystem as a whole if people noticed & reported those. I probably wouldn't mind dealing with that every now and then.

If we do want to avoid such breaks, I can see two fairly easy options:

2. We probably have admin rights on the Linux VM that AppVeyor spins up, so we could just uninstall Mono and reinstall a different version. That might be a fairly slow process, though.

3. We could prepare a Docker image with specific versions of Mono and the .NET Core SDK installed, then run our build inside a Docker container using that image. This would have to be updated by us every now and then.

I guess we could go with (1) and see how often Mono version updates actually break our build, and if that actually happens too often, still decide on (2) or (3) later on. What is your preference?

Reply:
Number 1 sounds good, you've convinced me that getting new Mono versions is probably good for us, we get new .NET Framework versions without any control anyway.

Mono 6 was out in June so it'll probably be pretty soon on AppVeyor.